java/src/Ice/src/main/java/com/zeroc/Ice/ConnectionI.java

```diff
@@ -492,6 +492,42 @@ public synchronized void asyncRequestCanceled(OutgoingAsyncBase outAsync, LocalE
     }
   }
 
+  private void sendResponse(OutgoingResponse response, boolean isTwoWay, byte compress) {
+    if (isTwoWay) {
+      //
+      // We may be executing on the "main thread" (e.g., in Android together with a custom
+      // dispatcher)
+      // and therefore we have to defer network calls to a separate thread.
+      //
+      final boolean queueResponse = _instance.queueRequests();
```

Comment:
This is copied straight from sendResponse below.

This needs to be cleaned up in a follow-up PR that eliminates ResponseHandler.

Reply:
Fixed by #2296.